subcommand/acl-controller/command.go

```diff
@@ -503,16 +503,36 @@ func (c *Command) upsertAnonymousTokenPolicy(consulClient *api.Client, agentConf
 		return fmt.Errorf("failed to list Consul datacenters: %w", err)
 	}
 
-	if !agentConfig.DebugConfig.MeshGatewayWANFederationEnabled || consulDC != primaryDC {
+	// Always configure the anonymous token. This is required for mesh-gateway traffic.
+	// For simplicity we configure this even if there are no mesh gateways in the datacenter.
+	if consulDC != primaryDC {
 		return nil
 	}
 
 	c.log.Info("Configuring anonymous token", "datacenter", consulDC, "primary-datacenter", primaryDC)
 
+	// This controller may not be running in the default partition.
+	// If the default partition is not an ECS cluster (and not a consul-k8s cluster)
+	// the anonymous token won't be configured correctly. In order to ensure that,
+	// we will always configure the anonymous token in the default partition so that
+	// mesh gateways actually work across partitions.
+	var qopts *api.QueryOptions
+	var wopts *api.WriteOptions
+	if c.flagPartitionsEnabled {
+		qopts = &api.QueryOptions{
+			Namespace: controller.DefaultPartition,
+			Partition: controller.DefaultNamespace,
+		}
+		wopts = &api.WriteOptions{
+			Namespace: controller.DefaultPartition,
+			Partition: controller.DefaultNamespace,
+		}
+	}
```

Comment:
Forcing the default partition here seems to match the k8s behavior:

https://github.com/hashicorp/consul-k8s/blob/aa05d3db2158e0887f49a0f5354893603ff6b6f2/control-plane/subcommand/server-acl-init/command.go#L479